src/contracts/atlas/DAppIntegration.sol

```diff
@@ -59,53 +45,85 @@ contract DAppIntegration {
         ATLAS = _atlas;
     }
 
+    // ---------------------------------------------------- //
+    //                DApp Integration Functions            //
+    // ---------------------------------------------------- //
+
     // Permissionlessly integrates a new dApp
     function initializeGovernance(address controller) external {
         address govAddress = IDAppControl(controller).getDAppSignatory();
-
         if (msg.sender != govAddress) revert AtlasErrors.OnlyGovernance();
 
-        bytes32 signatoryKey = keccak256(abi.encodePacked(msg.sender, controller, msg.sender));
+        // Add DAppControl gov as a signatory
+        bytes32 signatoryKey = keccak256(abi.encodePacked(controller, msg.sender));
```

Comment:
This feels potentially dangerous to me - we don't know that dappcontrol won't ever use delegateCall.  

I think we can add some extra protection by making sure that Atlas isn't in use (IE lock = 'unlocked') if we're going to allow the dappcontrol contracts to be their own signatories.

Reply:
Good idea, will add that check